### ISSUE TITLE
Fatal error: Unexpectedly found nil while unwrapping an Optional value

### ISSUE THREAD

**Issue by smhemel** [capacitor-community/date-picker#83]
**Describe the bug**
Date Picker is not showing up.
Getting Fatal error when I call atePicker.present() with valid options on Xcode.

My code:
```
DatePicker.present({
        mode: 'date',
        date: new Date().toISOString(),
        max: this.getMaxDate(from),
        theme: 'light',
      })
      .then((dateData) => {
         console.log(dateData);
      })
```

Error in Xcode:
`CapacitorCommunityDatePicker/Parse.swift:20: Fatal error: Unexpectedly found nil while unwrapping an Optional value`

<img width="1257" alt="Screenshot 2023-09-10 at 12 34 37 AM" src="https://github.com/capacitor-community/date-picker/assets/18726904/df6cd3ed-c549-48cc-9270-3d460a533bf6">

**Comment by DanielDarrenJones** [comment]
I'm getting this same issue, did you manage to find a solution @smhemel ? :)

**Comment by smhemel** [comment]
No, I didn't get any solution regarding this. That's why I have used another package instead of this. @DanielDarrenJones

**Comment by goodwin64** [comment]
We've faced the same error in our project recently. In our case we were passing the following data to the DatePicker:
```ts
let { value: newValue } = await DatePicker.present({
  mode, // "time"
  date: value || dateSuggestion || undefined, // 24-hour formatted time string
  timezone,
  format: dateFormat, // "HH:mm"
  is24h: true,
  ios: {
    timezone,
    format: dateFormat,
    style: 'wheels',
    mode,
  },
  android: {
    timezone,
    format: dateFormat,
    mode,
  },
});
```

Despite the fact we explicitly passed the `is24h` and `format` params the plugin used the device's platform built-in Date formatter (`DateFormatter` from the Foundation code) - so devices with the `12 hour` locale tried to parse the 24-hour formatted time, which returned `nil`. The things became worse because there's a "must-have-a-value" operator used:
```swift
// client/node_modules/@capacitor-community/date-picker/ios/Plugin/Parse.swift
return formatter.date(from: date)!;
//                               ^--- here
```

**Comment by Kamsou** [comment]
Same error

**Comment by smhemel** [comment]
> I'm getting this same issue, did you manage to find a solution @smhemel ? :)

No, I didn't find any solution. I have implemented a custom date-picker using <ion-datetime></ion-datetime> component.
![Screenshot 2024-05-14 234254](https://github.com/capacitor-community/date-picker/assets/18726904/6d056c82-a774-4f30-825e-1fd1491321ab)

**Comment by smhemel** [comment]
> Same error. @smhemel any idea ?

I didn't find any solution. Also, I did R&D to find an alternative date-picker plugin for the capacitor, but I didn't. 
After that, I have implemented a custom date-picker component using <ion-datetime></ion-datetime> component. Below, I have provided a screenshot. You can try the component.

![Screenshot 2024-05-14 234254](https://github.com/capacitor-community/date-picker/assets/18726904/6d056c82-a774-4f30-825e-1fd1491321ab)

**Comment by elmekkaoui-dev** [comment]
I had the same problem setting the format and the max properties as well to ISO resolve my issue
Here's an updated version of the DatePickerOptions I am using 
```
export const defaultPickerOptions: DatePickerOptions  = {
  locale: 'fr',
  max: new Date().toISOString(),
  mode: 'date',
  theme: 'dark',
  format: "yyyy-MM-dd'T'HH:mm:ss.sssZ"
}
```
